statsmodels/discrete/discrete_model.py

```diff
@@ -2371,11 +2371,10 @@ def __init__(self, model, mlefit, cov_type='nonrobust', cov_kwds=None,
 
 
     def __getstate__(self):
-        try:
-            #remove unpicklable callback
-            self.mle_settings['callback'] = None
-        except (AttributeError, KeyError):
-            pass
+        # remove unpicklable methods
+        self.mle_settings['callback'] = None
```

Comment:
I think, catching the AttributeError was added in case `mle_settings` itself also doesn't exist.
(I think OLS and similar linear algebra models don't have it but the unit tests don't fail (?).)